### DISCUSSION TITLE
Prevent imported modules of a mocked file to be included in Module Graph

### DISCUSSION THREAD

**Discussion by pll-skytech** [vitest-dev/vitest#7091] [Q&A]
I am struggling to optimise some of my tests and can't seem to find any solution to my problem. Here is a simplification:
I have fileA.tsx that `imports { x } from 'path/fileB'` and in the test file i use `vi.mock('path/fileB', () => ({ x: vi.fn() })`.

But i dont understand why in vitest UI Module Graph i see all the nested imports of fileB (it has a few imports that imports other things so i end up with a quite big graph for nothing)... Is there a way i can tell vitest that it should not import any code from fileB since it will be mocked and i do not need any code from it ?

Thanks for any help!

**Comment by hi-ogawa** [comment]
Can you provide a reproduction? 
From my testing https://stackblitz.com/edit/vitest-dev-vitest-7c7ra8b3?file=test%2Fbasic.test.ts, it doesn't look like nested imports from mocked module shows up in module graph ui.

<details><summary>Screenshot</summary>

![image](https://github.com/user-attachments/assets/e0ab9e0e-eb2c-4470-9216-affc665d3e74)

</details>

**Reply by pll-skytech** [reply]
Unfortunately I am unable to reproduce, it does work as intended on a basic project with same config/setup as my current project. Even on my current project it works correctly on most files, it seems to be only happening for a specific files that they pull all dependencies even if i comment out all the code, import some dummy files with dummy functions... I commented out my vite config to make it very basic, i just can't put my finger on what differs in my current project that I can't share..

Thanks for the quick reply. The only thing that could help me at this point would be a way to debug the code that determines when something is included or not so I could maybe track what's going on....
